pkg/driver/krun/krun_driver_darwin.go

```diff
@@ -0,0 +1,203 @@
+// SPDX-FileCopyrightText: Copyright The Lima Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package krun
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/lima-vm/lima/v2/pkg/driver"
+	"github.com/lima-vm/lima/v2/pkg/executil"
+	"github.com/lima-vm/lima/v2/pkg/limatype"
+	"github.com/lima-vm/lima/v2/pkg/limatype/filenames"
+	"github.com/lima-vm/lima/v2/pkg/ptr"
+)
+
+type LimaKrunDriver struct {
+	Instance     *limatype.Instance
+	SSHLocalPort int
+
+	krunCmd    *exec.Cmd
+	krunWaitCh chan error
+}
+
+var _ driver.Driver = (*LimaKrunDriver)(nil)
+
+func New() *LimaKrunDriver {
+	return &LimaKrunDriver{}
+}
+
+func (l *LimaKrunDriver) Configure(inst *limatype.Instance) *driver.ConfiguredDriver {
+	l.Instance = inst
+	l.SSHLocalPort = inst.SSHLocalPort
+
+	return &driver.ConfiguredDriver{
+		Driver: l,
+	}
+}
+
+func (l *LimaKrunDriver) CreateDisk(ctx context.Context) error {
+	return EnsureDisk(ctx, l.Instance)
+}
+
+func (l *LimaKrunDriver) Start(ctx context.Context) (chan error, error) {
+	krunCmd, err := Cmdline(l.Instance)
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct krunkit command line: %w", err)
+	}
+	krunCmd.SysProcAttr = executil.BackgroundSysProcAttr
+
+	logPath := filepath.Join(l.Instance.Dir, "krun.log")
+	logfile, err := os.OpenFile(logPath, os.O_CREATE|os.O_WRONLY|os.O_APPEND, 0644)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open krunkit logfile: %w", err)
+	}
+	krunCmd.Stderr = logfile
+
+	logrus.Infof("Starting krun VM (hint: to watch the progress, see %q)", logPath)
+	logrus.Debugf("krunCmd.Args: %v", krunCmd.Args)
+
+	if err := krunCmd.Start(); err != nil {
+		logfile.Close()
+		return nil, err
+	}
+
+	pidPath := filepath.Join(l.Instance.Dir, filenames.PIDFile(*l.Instance.Config.VMType))
+	if err := os.WriteFile(pidPath, []byte(fmt.Sprintf("%d\n", krunCmd.Process.Pid)), 0644); err != nil {
+		logrus.WithError(err).Warn("Failed to write PID file")
+	}
+
+	l.krunCmd = krunCmd
+	l.krunWaitCh = make(chan error, 1)
+	go func() {
+		defer func() {
+			logfile.Close()
+			os.RemoveAll(pidPath)
+			close(l.krunWaitCh)
+		}()
+		l.krunWaitCh <- krunCmd.Wait()
+	}()
+
+	return l.krunWaitCh, nil
+}
+
+func (l *LimaKrunDriver) Stop(ctx context.Context) error {
+	if l.krunCmd == nil {
+		return nil
+	}
+
+	if err := l.krunCmd.Process.Signal(os.Interrupt); err != nil {
+		logrus.WithError(err).Warn("Failed to send interrupt signal")
+	}
+
+	timeout := time.After(30 * time.Second)
+	select {
+	case <-l.krunWaitCh:
+		return nil
+	case <-timeout:
+		return l.krunCmd.Process.Kill()
+	}
```

Comment:
You need to wait for the process after killing it. Otherwise it will remain a zombie until this process terminates.